### ISSUE TITLE
Restangular and usage of Resource's links

### ISSUE THREAD

**Issue by dragonflypl** [mgonto/restangular#1370]
I'm consuming RESTful API with HATEOAS. It's pretty decent and describes resources completely.

Restangular is pretty good at chaining and constructing urls e.g.: `Restangular.one('profiles', 'standard-profile').all('rules')`

will generate: `profiles/standard-profile/rules.`

Because my profile resource already has all the related links included as metadata, I would like to utilize them, i.e. I don't want to construct queries manually with (and Restangular will concat url behind the scenes):

`Restangular.one('profiles', 'standard-profile').all('rules')`

This is preety simple example, in reality my object has plenty of links as metadata and these links have query strings and are more complicated.

Instead I would like to tell Restangular - if rules (or whatever is available in links object on resource level) are to be fetched, use url included as links.rules. It's something similar to selfLink.

Is is possible with Restangular?

Basically this questions could be also rephrased: do you know any approach to use HATEOAS self navigating features using links with Restangular without writing too much code.

For instance this is something I came up with:

```
Restangular.addElementTransformer('movie', false, function(movie) {
    movie.theSameGenreMovies = function() {
        return Restangular.allUrl('movie', movie.links.genreMovies).getList();
    }

    // this does not work as expected
    //movie.addRestangularMethod('theSameGenreMovies', 'getList', genresLink, undefined);

    return movie;
});
```

**Comment by mappedinn** [comment]
:+1:

**Comment by daviesgeek** [comment]
That looks like a good way to do it. I'm not super familiar with HATEOAS, but your example looks to be the best way to do it. I'm confused as to why the second part of your example doesn't work:

>     // this does not work as expected
    //movie.addRestangularMethod('theSameGenreMovies', 'getList', genresLink, undefined);

What are you trying to accomplish? `genresLink` is undefined, so it won't work. Change it to this instead and I believe it should work:

```js
movie.addRestangularMethod('theSameGenreMovies', 'getList', movie.links. genreMovies);
```

**Comment by dragonflypl** [comment]
Yeap, code sample was incomplete. But even when done like this:

`movie.addRestangularMethod('theSameGenreMovies_2', 'getList', movie.links.genreMovies);`

it does not work, because `movie.links.genreMovies` has complete url, which is then appended to current resource's url. As a result, complete request is as follows:

`http://api.themoviedb.org/3/movie/2661/http://api.themoviedb.org/3/genre/12/movies?api_key=e5b46b96a0d706e32df7117c8ac68e8f`

**Comment by daviesgeek** [comment]
Unfortunately I'm not sure of a way to support this, other than manually removing the beginning part of the URL in `movie.links.genreMovies`.
I'd do something like this:

```js
movie.addRestangularMethod('theSameGenreMovies_2', 'getList', movie.links.genreMovies.replace(baseURL, '');
```
